From 1111111111111111111111111111111111111111 Mon Sep 17 00:00:00 2001
From: Anton <anton@example.com>
Date: Mon, 23 Mar 2026 00:00:00 +0000
Subject: [PATCH] version: override mpv version string

---
 MPV_VERSION | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 MPV_VERSION

diff --git a/MPV_VERSION b/MPV_VERSION
new file mode 100644
index 0000000000..1111111111
--- /dev/null
+++ b/MPV_VERSION
@@ -0,0 +1 @@
+EzTvLibA-1.0-0.41.0
--
2.43.0
