diff --git a/libavcodec/ccaption_dec.c b/libavcodec/ccaption_dec.c
index 6f4d4fca41..29a969aa9e 100644
--- a/libavcodec/ccaption_dec.c
+++ b/libavcodec/ccaption_dec.c
@@ -390,7 +390,8 @@ static int validate_cc_data_pair(const uint8_t *cc_data_pair, uint8_t *hi)
             return AVERROR_INVALIDDATA;
         }
         if (!av_parity(cc_data_pair[1])) {
-            *hi = 0x7F;
+            //*hi = 0x7F;
+            return AVERROR_INVALIDDATA;
         }
     }
 
